src/java_bytecode/java_utils.h

```diff
@@ -66,4 +66,6 @@ irep_idt resolve_friendly_method_name(
 /// \param type: expected result type (typically expr.type().subtype())
 dereference_exprt checked_dereference(const exprt &expr, const typet &type);
 
+size_t find_closing_delimiter(const std::string &, size_t, char, char);
```

Comment:
Nit: name the parameters so users of the function can know which char does what. You might also consider overloading this with a method that takes just one char (say '(') and figure out the closing one.